src/applications/travel-pay/util/dates.js

```diff
@@ -7,3 +37,63 @@ export function formatDateTime(datetimeString) {
 
   return [formattedDate, formattedTime];
 }
+
+/**
+ * Returns an array of the following date range filters:
+ *  - The past 3 months from today's date
+ *  - The previous full quarters (e.g. if today falls in Q2 2024,
+ *  return Q1 2024, Q4 2023, and Q3 2023 )
+ *  - All of the current year to date
+ *  - All of the previous year
+ */
+export function getDateFilters() {
+  const today = utcToZonedTime(new Date());
+  let quarter = getQuarter(today);
+
+  const dateRanges = [];
+
+  dateRanges.push({
+    label: 'Past 3 Months',
+    start: subMonths(today, 3),
+    end: today,
+  });
+
+  // Calculate the last 3 complete quarters
+  for (let i = 1; i < 4; i++) {
+    quarter -= 1;
+    if (quarter < 1) {
+      quarter = 4;
+    }
+
+    const quarterStart = startOfQuarter(subQuarters(today, i));
+    const quarterEnd = endOfQuarter(quarterStart);
+    const quarterYear = getYear(quarterStart);
+
+    const startMonth = months.find(
+      month => month.value === getMonth(quarterStart) + 1,
+    );
+    const endMonth = months.find(
+      month => month.value === getMonth(quarterEnd) + 1,
+    );
+
+    dateRanges.push({
+      label: `${startMonth.label} ${quarterYear} - ${
+        endMonth.label
+      } ${quarterYear}`,
+      start: new Date(quarterYear, startMonth.value - 1, 1),
+      end: endOfMonth(new Date(quarterYear, endMonth.value - 1)),
+    });
+  }
```

Comment:
Thinking out loud here - it seems like we could increase the clarity of `getDateFilters()` with something like:

```javascript
const dateRangeLabel = (start, end) => {
  return `${format(quarterStart, 'MMM yyyy')} - ${format(quarterEnd, 'MMM yyyy'}`
}

dateRanges.push({
  label: dateRangeLabel(quarterStart, quarterEnd),
  start: quarterStart,
  end: quarterEnd,
});
```

Reply:
What are your thoughts? I think this would remove the need for a lot of intermediary variables and structure, like the lookup table defined above could go away with this.

Reply:
Good call. I was definitely overusing variables a bit. Cleaned it up and it's a lot more straightforward now